tests/test_templates_embeddings.py

```diff
@@ -0,0 +1,408 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the :mod:`pennylane.templates.utils` module.
+"""
+# pylint: disable=protected-access,cell-var-from-loop
+import pytest
+from math import pi
+import numpy as np
+import pennylane as qml
+from pennylane.templates.embeddings import (AngleEmbedding,
+                                            BasisEmbedding,
+                                            AmplitudeEmbedding,
+                                            DisplacementEmbedding,
+                                            SqueezingEmbedding)
+from pennylane import Beamsplitter
+
+
+@pytest.fixture(scope="session",
+                params=[1, 2, 5])
+def n_subsystems(request):
+    """Number of qubits or modes."""
+    return request.param
+
+
+@pytest.fixture(scope="session")
+def qubit_device(n_subsystems):
+    """Number of qubits or modes."""
+    return qml.device('default.qubit', wires=n_subsystems)
+
+
+@pytest.fixture(scope="session")
+def gaussian_device(n_subsystems):
+    """Number of qubits or modes."""
+    return qml.device('default.gaussian', wires=n_subsystems)
+
+
+class TestAngleEmbd:
+    """ Tests the pennylane.templates.embeddings.AngleEmbedding method."""
+
+    def test_angle_embedding_state_rotx(self, qubit_device, n_subsystems):
+        """Checks the state produced by pennylane.templates.embeddings.AngleEmbedding()
+           using the rotation='X' strategy."""
+
+        features = [pi/2,  pi/2, pi/4, 0]
+
+        @qml.qnode(qubit_device)
+        def circuit(x=None):
+            AngleEmbedding(features=x, n_wires=n_subsystems, rotation='X')
+            qml.PauliX(wires=0)
+            AngleEmbedding(features=x, n_wires=n_subsystems, rotation='X')
+            return [qml.expval.PauliZ(i) for i in range(n_subsystems)]
+
+        res = circuit(x=features[:n_subsystems])
+        target = [1, -1, 0, 1, 1]
+
+        assert np.allclose(res, target[:n_subsystems])
+
+    def test_angle_embedding_exception_rotx(self):
+        """Verifies that pennylane.templates.embeddings.AngleEmbedding()
+           using the rotation='X' strategy raises an exception if there are fewer
+           rotation gates than features."""
+
+        features = [0, 0, 0, 0]
+        n_subsystems = 1
+        dev = qml.device('default.qubit', wires=n_subsystems)
+
+        @qml.qnode(dev)
+        def circuit(x=None):
+            AngleEmbedding(features=x, n_wires=n_subsystems, rotation='X')
+            qml.PauliX(wires=0)
+            AngleEmbedding(features=x, n_wires=n_subsystems, rotation='X')
+            return [qml.expval.PauliZ(i) for i in range(n_subsystems)]
+
+        with pytest.raises(ValueError) as excinfo:
+            circuit(x=features)
+        assert excinfo.value.args[0] == 'Number of features to embed cannot be larger than num_wires, but is 4.'
```

Comment:
I used your (more elegant) version, except from one place. Is seems it cannot handle the string 2**n for some reason, so that's where I left the previous one.